cmd/goal/node.go

```diff
@@ -119,18 +122,55 @@ var nodeCmd = &cobra.Command{
 	},
 }
 
+func getMissingCatchpointLabel(URL string) (label string, err error) {
+	resp, err := http.Get(URL)
+	if err != nil {
+		return
+	}
+	if resp.StatusCode != 200 {
+		err = errors.New(resp.Status)
+		return
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return
+	}
+	label = string(body)
+	label = strings.TrimSuffix(label, "\n")
+
+	// check if label is a valid catchpoint label
+	_, _, err = ledgercore.ParseCatchpointLabel(label)
+	if err != nil {
+		return
+	}
+	return
+}
+
 var catchupCmd = &cobra.Command{
 	Use:     "catchup",
 	Short:   "Catchup the Algorand node to a specific catchpoint",
-	Long:    "Catchup allows making large jumps over round ranges without the need to incrementally validate each individual round.",
+	Long:    "Catchup allows making large jumps over round ranges without the need to incrementally validate each individual round. If no catchpoint is provided, this command attempts to lookup the latest catchpoint from algorand-catchpoints.s3.us-east-2.amazonaws.com.",
 	Example: "goal node catchup 6500000#1234567890ABCDEF01234567890ABCDEF0\tStart catching up to round 6500000 with the provided catchpoint\ngoal node catchup --abort\t\t\t\t\tAbort the current catchup",
 	Args:    catchpointCmdArgument,
 	Run: func(cmd *cobra.Command, args []string) {
-		if abortCatchup == false && len(args) == 0 {
-			fmt.Println(errorCatchpointLabelMissing)
-			os.Exit(1)
-		}
-		onDataDirs(func(datadir string) { catchup(datadir, args) })
+		onDataDirs(func(dataDir string) {
+			if !abortCatchup && len(args) == 0 {
+				client := ensureAlgodClient(dataDir)
+				vers, err := client.AlgodVersions()
+				if err != nil {
+					reportErrorf(errorNodeStatus, err)
+				}
+				genesis := strings.Split(vers.GenesisID, "-")[0]
+				URL := fmt.Sprintf(catchpointURL, genesis)
+				label, err := getMissingCatchpointLabel(URL)
+				if err != nil {
+					fmt.Println(errorUnableToLookupCatchpointLabel)
```

Comment:
Building off of what @winder said earlier, return the original error wrapped in the new text. e.g:

`fmt.Errorf("Unable to fetch catchpoint label: %w", err)`